src/System.Net.Http/tests/UnitTests/HPack/HPackRoundtripTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http.HPack;
+using System.Net.Http.Headers;
+using System.Text;
+using Xunit;
+
+namespace System.Net.Http.Unit.Tests.HPack
+{
+    public class HPackRoundtripTests
+    {
+
+        #region "tests"
+
+        // TODO: Extend to Property-Based Test using FsCheck 
+
+        [Fact]
+        public void HPack_Roundtrip_Single_Header_With_Single_Value()
```

Comment:
@davidsh has specific desires for how test names are formed.  I'll let him comment if he'd like to see something different here.

Reply:
@eiriktsarpalis Please look at existing HttpClient tests for example of naming pattern.  We use a 3-part name consisting of:

```
<property_or_method_under_test>_<conditions>_<expectedresult>
```

We've used this [book](https://smile.amazon.com/Art-Unit-Testing-examples/dp/1617290890/ref=sr_1_3?keywords=the+art+of+unit+testing&qid=1566428801&s=gateway&sr=8-3), "The Art of Unit Testing" as the blueprint for our testing methodology:

Reply:
How about this? `HPack_HeaderEncodeDecodeRoundtrip_ShouldMatchOriginalInput`

Reply:
How about this? `HPack_HeaderEncodeDecodeRoundtrip_ShouldMatchOriginalInput`